docs: add threat model to SECURITY.md (#2360)
pinojs/pino: SECURITY.md

This document describes the management of vulnerabilities for the
Pino project and all modules within the Pino organization.

## The Pino Threat Model

Pino is a fast JSON logger for Node.js. Understanding what Pino considers
a security vulnerability requires understanding its trust boundaries.

Pino's threat model builds upon the
[Node.js threat model](https://github.com/nodejs/node/blob/main/SECURITY.md#the-nodejs-threat-model).
We recommend reading that document first, as Pino inherits its trust assumptions.

> Pino trust the applications and the environment that it is being run on top.

Reviewer: ```suggestion
Pino trusts the applications using it and the environment that it is running in.
```

This document describes the management of vulnerabilities for the
Pino project and all modules within the Pino organization.

## The Pino Threat Model

Pino is a fast JSON logger for Node.js. Understanding what Pino considers
a security vulnerability requires understanding its trust boundaries.

Pino's threat model builds upon the
[Node.js threat model](https://github.com/nodejs/node/blob/main/SECURITY.md#the-nodejs-threat-model).
We recommend reading that document first, as Pino inherits its trust assumptions.

### What Pino Trusts

The following are considered trusted and are outside the scope of Pino's
security model:

- **Application code**: Pino trusts the application code that invokes
  logging functions. If your code logs sensitive data (passwords, tokens,
  PII), that is an application-level concern, not a Pino vulnerability.
- **Log message content**: Data passed to `logger.info()`, `logger.error()`,
  etc. is trusted. Use [serializers](./docs/api.md#opt-serializers) and the
  [redact](./docs/redaction.md) feature to sanitize sensitive data before it
  is written to the log destination.
- **Configuration options**: Options passed to `pino()` are trusted.

>   Misconfiguration (e.g., insecure file paths, unsafe serializers) is

Reviewer: ```suggestion
  Misconfiguration (e.g. insecure file paths or unsafe serializers) is
```
